libr/anal/p/anal_xtensa.c

```diff
@@ -634,6 +634,12 @@ static inline void sign_extend(st32 *value, ut8 bit) {
 	}
 }
 
+static inline void sign_extend2(st32 *value, ut8 bit1, ut8 bit2, ut8 shift) {
+	ut8 bit = ((*value >> bit1) & 1) & ((*value >> bit2) & 1);
```

Comment:
this is bool

Reply:
you can save a lilne andd just if ((*vallue..

Reply:
Actually not bool due to only bitwise operations used here. Will be bool if && used instead &.